tests/test_check.py

```diff
@@ -127,6 +127,32 @@ def test_check_no_description(monkeypatch, capsys):
     )
 
 
+def test_strict_fails_on_warnings(monkeypatch, capsys):
```

Comment:
I took the liberty of moving this to be near similar tests, instead of at the end of the file.